### PR TITLE
Add flash attention backward grad check

### PR DESCRIPTION
**Pull request by iosmers**
### PR types

Others
### PR changes

Others
### Description

Add backward checn for flash attn with mask 
Pcard-70459

### CODE REVIEW

File: test/legacy_test/test_flash_attention.py [hunk 1]
```diff
@@ -312,7 +312,7 @@ def test_all(self):
     not core.is_compiled_with_cuda()
     or get_cuda_version() < 11040
     or not is_sm_supported,
-    "core is not compiled with CUDA and cuda version need larger than or equal to 11.3"
+    "core is not compiled with CUDA and cuda version need larger than or equal to 11.4"
     "and device's compute capability must be 7.5 or 8.x",
```
Comment (umiswing) [inline test/legacy_test/test_flash_attention.py:316]: 这个提升信息应该改成```8.x or 90```了

File: python/paddle/nn/functional/flash_attention.py [hunk 2]
```diff
@@ -417,6 +417,7 @@ def scaled_dot_product_attention(
     dropout_p=0.0,
     is_causal=False,
     training=True,
+    name=None,
```
Comment (umiswing) [inline python/paddle/nn/functional/flash_attention.py:420]: 新增的```name```这个参数的作用是什么呢？

Comment (iosmers) [reply python/paddle/nn/functional/flash_attention.py:420]: 为了与其他API兼容，详见https://github.com/PaddlePaddle/Paddle/pull/55242#pullrequestreview-1556583112

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Review by jeff41404 — APPROVED** [review]
LGTM

File: test/legacy_test/test_flash_attention.py (hunk 1 above)
Comment (iosmers) [reply test/legacy_test/test_flash_attention.py:316]: 待FA-2支持attn_mask后修改

File: paddle/phi/kernels/gpu/flash_attn_grad_kernel.cu [hunk 3]
```diff
@@ -124,15 +128,14 @@ void FlashAttnUnpaddedGradImpl(const Context& ctx,
       mask_dims.data() ? mask_dims.data() : nullptr,
       nullptr);
   CheckFlashAttnStatus(succ);
-
```
Comment (Xreki) [inline paddle/phi/kernels/gpu/flash_attn_grad_kernel.cu:127]: 代码中应适当地加一些空行，便于分段阅读。

**Review by Xreki — APPROVED** [review]
LGTM. 有两个建议：
- PR title应准确描述PR的功能，建议完善下
- 若有文档修改，后续需要将文档预览效果贴在PR描述中，可从PR-CI-Paddle-Doc-Preview CI中找到PR自动文档预览的链接，如下。
image

**Review by umiswing — APPROVED** [review]
LGTM

**Comment by iosmers** [comment]
文档预览结果
````>>> import paddle
>>> q = paddle.rand((1, 128, 2, 16), dtype=paddle.bfloat16)
>>> output = paddle.nn.functional.scaled_dot_product_attention(q, q, q, None, 0.9, False)
>>> print(output)
````
